source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
+    }
+
+    public static String getProtocolHandlerProject() {
+        return System.getProperty(PROTOCOL_HANDLER_PROJECT_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerBranch() {
+        return System.getProperty(PROTOCOL_HANDLER_BRANCH_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerRepository() {
+        return System.getProperty(PROTOCOL_HANDLER_REPOSITORY_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerUrl() {
+        return System.getProperty(PROTOCOL_HANDLER_URL_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerEncodedUrl() {
+        return URIUtils.newURI(System.getProperty(PROTOCOL_HANDLER_URL_PROPERTY, StringUtil.EMPTY)).toASCIIString();
+    }
+
+    private static boolean tryParseGitRepoUrl(final String repoUrl) {
+        final URI uri = URIUtils.newURI(repoUrl);
+        final String[] pathItems = uri.getPath().split("/"); //$NON-NLS-1$
+        int n = pathItems.length;
+
+        if (n < 2 || !GIT_TOKEN.equals(pathItems[n - 2])) {
+            return false;
+        }
+
+        final String repository = pathItems[n - 1];
+        String path;
+        String project;
+
+        // Note that because uri.getPath() is absolute, pathItems[0] is always
+        // an empty string.
+        if (ServerURIUtils.isHosted(uri)) {
+            /*
+             * @formatter off            
+             * Possible URLs are:
+             * https://account.visualstudio.com/_git/repository                           (n=3)
+             * https://account.visualstudio.com/project/_git/repository                   (n=4) 
+             * https://account.visualstudio.com/DefaultCollection/_git/repository         (n=4) 
+             * https://account.visualstudio.com/DefaultCollection/project/_git/repository (n=5) 
+             * in the future, "DefaultCollection" will be replaced with "Organization"
+             * @formatter:on          
+             */
+            switch (n) {
+                case 5:
+                    project = pathItems[2];
+                    path = '/' + pathItems[1];
+                    break;
+                case 4:
+                    if (DEFAULT_COLLECTION_TOKEN.equalsIgnoreCase(pathItems[1])) {
+                        project = repository;
+                        path = '/' + pathItems[1];
+                    } else {
+                        project = pathItems[1];
+                        path = null;
+                    }
+                    break;
+                case 3:
+                    project = repository;
+                    path = null;
+                    break;
+                default:
+                    return false;
+            }
+        } else {
+            /*
+             * @formatter off            
+             * Possible URLs are:
+             * https://server:port/path/collection/_git/repository                           (n=5)
```

Comment:
Don't forget about the case where path doesn't exist which would make this a lot more complicated.